modules/planning/tasks/optimizers/open_space_trajectory_partition/open_space_trajectory_partition.cc

```diff
@@ -17,21 +17,409 @@
+    } else if (init_direction < 0) {
+      ADEBUG << "initial speed oscillate too "
+                "frequent around zero";
+      (*current_trajectory).second = canbus::Chassis::GEAR_REVERSE;
+    } else {
+      ADEBUG << "Invalid trajectory start! initial speeds too small to decide "
+                "gear";
+      (*current_trajectory).second = canbus::Chassis::GEAR_DRIVE;
+    }
+  }
+
+  // Partition trajectory points into each trajectory
+  for (size_t i = 0; i < horizon; ++i) {
+    // shift from GEAR_DRIVE to GEAR_REVERSE if v < 0
+    // then add a new trajectory with GEAR_REVERSE
+    if (interpolated_trajectory_result_ptr->at(i).v() < -kepsilon &&
+        current_trajectory->second == canbus::Chassis::GEAR_DRIVE) {
+      paritioned_trajectories_ptr->emplace_back();
+      current_trajectory = &(paritioned_trajectories_ptr->back());
+      (*current_trajectory).second = canbus::Chassis::GEAR_REVERSE;
+      distance_s = 0.0;
+    }
+    // Shift from GEAR_REVERSE to GEAR_DRIVE if v > 0
+    // then add a new trajectory with GEAR_DRIVE
+    if (interpolated_trajectory_result_ptr->at(i).v() > kepsilon &&
+        current_trajectory->second == canbus::Chassis::GEAR_REVERSE) {
+      paritioned_trajectories_ptr->emplace_back();
+      current_trajectory = &(paritioned_trajectories_ptr->back());
+      (*current_trajectory).second = canbus::Chassis::GEAR_DRIVE;
+      distance_s = 0.0;
+    }
+
+    (*current_trajectory).first.emplace_back();
+    common::TrajectoryPoint* point = &(current_trajectory->first.back());
+    point->set_relative_time(
+        interpolated_trajectory_result_ptr->at(i).relative_time());
+    point->mutable_path_point()->set_x(
+        interpolated_trajectory_result_ptr->at(i).path_point().x());
+    point->mutable_path_point()->set_y(
+        interpolated_trajectory_result_ptr->at(i).path_point().y());
+    point->mutable_path_point()->set_theta(
+        interpolated_trajectory_result_ptr->at(i).path_point().theta());
+    if (i > 0) {
+      distance_s +=
+          (current_trajectory->second == canbus::Chassis::GEAR_REVERSE ? -1.0
+                                                                       : 1.0) *
+          std::sqrt(
+              (interpolated_trajectory_result_ptr->at(i).path_point().x() -
+               interpolated_trajectory_result_ptr->at(i - 1).path_point().x()) *
+                  (interpolated_trajectory_result_ptr->at(i).path_point().x() -
+                   interpolated_trajectory_result_ptr->at(i - 1)
+                       .path_point()
+                       .x()) +
+              (interpolated_trajectory_result_ptr->at(i).path_point().y() -
+               interpolated_trajectory_result_ptr->at(i - 1).path_point().y()) *
+                  (interpolated_trajectory_result_ptr->at(i).path_point().y() -
+                   interpolated_trajectory_result_ptr->at(i - 1)
+                       .path_point()
+                       .y()));
```

Comment:
path_point_i = interpolated_trajectory_result_ptr->at(i).path_point();
path_point_pre_i = interpolated_trajectory_result_ptr->at(i - 1).path_point();
Then make it shorter and readable.